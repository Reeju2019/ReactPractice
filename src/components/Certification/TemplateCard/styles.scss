.template-card {
  border: solid 3px #f5f4f8;
}

.template-caption {
  font-size: 0.9rem;
  font-weight: 500;
  opacity: 0.9;
  color: #6851b5;
}

.template-card-selected {
  border: solid 3px #6851b5;
}
