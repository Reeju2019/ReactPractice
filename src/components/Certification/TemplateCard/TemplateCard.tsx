import React from "react";
import { Col, FormLabel, Card } from "react-bootstrap";
import "./styles.scss";

/**
 * @description Props of Template Card Component
 * @memberof TemplateCard
 * @property {string} id Template id
 * @property {string} imgUrl Template image url
 * @property {string} caption Set the Template caption
 * @property {any} setActiveCardRef Set active card ref
 */
/* eslint-disable */
interface IProps {
  id: string;
  imgUrl: string;
  caption: string;
  setActiveCardRef: any;
}

/**
 * @description States of Template Card Component
 * @memberof TemplateCard
 */
/* eslint-disable */
interface IState {}

/**
 * @component
 * This component was used in Certificate component
 * @description TemplateCard Component
 */
class TemplateCard extends React.Component<IProps, IState> {
  imgRef: any;

  /**
   * @constructor
   * @description Constructor
   * @param {IProps} props Props
   * @returns void
   */
  constructor(props: IProps) {
    super(props);
    this.imgRef = React.createRef();
    this.setCardSelectedStyle = this.setCardSelectedStyle.bind(this);
  }

  /**
   * Set Card Selected Style
   * @property {Function} setCardSelectedStyle Set card selected style
   * @returns {void}
   */
  setCardSelectedStyle() {
    this.props.setActiveCardRef(this.imgRef);
  }

  /**
   * Lifecycle method componentDidMount()
   * @property {Function} componentDidMount Component Did Mount
   * @returns {void}
   */
  componentDidMount(): void {
    if (this.props.id == "1") {
      this.props.setActiveCardRef(this.imgRef);
    }
  }

  render() {
    return (
      <>
        <Col xs={4} md={4} lg={3}>
          <Card className="me-3">
            <Card.Img
              className="template-card"
              id={this.props.id}
              variant="top"
              src={this.props.imgUrl}
              onClick={this.setCardSelectedStyle}
              ref={this.imgRef}
            />
          </Card>
          <FormLabel className="template-caption">
            {this.props.caption}
          </FormLabel>
        </Col>
      </>
    );
  }
}

export default TemplateCard;
export { TemplateCard };
