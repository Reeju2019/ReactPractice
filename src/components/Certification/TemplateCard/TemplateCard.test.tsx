import { shallow } from "enzyme"
import ReactDOM from "react-dom"
import TemplateCard from "./TemplateCard"

/**
 * Unit testing for TemplateCard
 */
describe("Template Card Testing", () => {
  it("render without crashing", () => {
    const div = document.createElement("div")
    ReactDOM.render(
      <TemplateCard
        id="1"
        imgUrl=""
        caption="caption"
        setActiveCardRef={() => {}}
      />,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
  it("should render initial layout", () => {
    const component = shallow(
      <TemplateCard
        id="1"
        imgUrl=""
        caption="caption"
        setActiveCardRef={() => {}}
      />
    )
    expect(component.getElements()).toMatchSnapshot()
  })
})
