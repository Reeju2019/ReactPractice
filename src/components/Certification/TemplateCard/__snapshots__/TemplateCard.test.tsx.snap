// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Template Card Testing should render initial layout 1`] = `
Array [
  <React.Fragment>
    <Col
      lg={3}
      md={4}
      xs={4}
    >
      <Card
        body={false}
        className="me-3"
      >
        <CardImg
          className="template-card"
          id="1"
          onClick={[Function]}
          src=""
          variant="top"
        />
      </Card>
      <FormLabel
        className="template-caption"
        column={false}
        visuallyHidden={false}
      >
        caption
      </FormLabel>
    </Col>
  </React.Fragment>,
]
`;
